### ISSUE TITLE
InteractivePasteCommandhandlerTests.PasteCommandWithInteractiveFormat failure

### ISSUE THREAD

**Issue by 333fred** [dotnet/roslyn#25315]
`InteractivePasteCommandhandlerTests.PasteCommandWithInteractiveFormat` failed in the signed build, https://devdiv.visualstudio.com/DevDiv/_build/index?buildId=1450639&_a=summary&tab=details. Unfortunately test results for this test seem to be missing (though the failure is in the console output and failed the build), so I don't have a stack trace. Possibly related to https://github.com/dotnet/roslyn/issues/20840.

**Comment by 333fred** [comment]
Also failed in CI for dev15.7.x-preview1, this time with stack. @dpoeschl there appears to be some debug asserts in the RenameTrackingTaggerProvider.
https://ci.dot.net/job/dotnet_roslyn/view/Official%20Builds/job/dev15.7-preview1-vs-deps/job/windows_debug_unit32/1/
```
Microsoft.CodeAnalysis.Editor.UnitTests.InteractivePaste.InteractivePasteCommandhandlerTests.PasteCommandWithInteractiveFormat
      System.AggregateException : 5 exception(s) were thrown during test.
      Note: exceptions may have been thrown by another test running concurrently with
      this test.  This can happen with any tests that share the same ExportProvider.
      Examining individual exception stacks may help reveal the original test and source 
      of the problem.
      
      Exception 0:
      Microsoft.CodeAnalysis.ThrowingTraceListener+DebugAssertFailureException: When created kind       : StaUnitTest
      When created thread id  : 23
      When created thread name: StaTaskScheduler thread
      Current thread id       : 4
      Current thread name     : 
      
         at Microsoft.CodeAnalysis.ThrowingTraceListener.Fail(String message, String detailMessage)
         at System.Diagnostics.TraceListener.Fail(String message)
         at System.Diagnostics.TraceInternal.Fail(String message)
         at System.Diagnostics.Debug.Assert(Boolean condition, String message)
         at Microsoft.CodeAnalysis.Editor.Shared.Utilities.ForegroundThreadAffinitizedObject.AssertIsForeground()
         at Microsoft.CodeAnalysis.Editor.Implementation.RenameTracking.RenameTrackingTaggerProvider.StateMachine.Connect()
         at Microsoft.CodeAnalysis.Editor.Implementation.RenameTracking.RenameTrackingTaggerProvider.Tagger..ctor(StateMachine stateMachine, ITextUndoHistoryRegistry undoHistoryRegistry, IWaitIndicator waitIndicator, IEnumerable`1 refactorNotifyServices)
         at Microsoft.CodeAnalysis.Editor.Implementation.RenameTracking.RenameTrackingTaggerProvider.CreateTagger[T](ITextBuffer buffer)
         at Microsoft.VisualStudio.Text.Tagging.Implementation.TagAggregator`1.GatherTaggers(ITextBuffer textBuffer)
      
      Exception 1:
      Microsoft.CodeAnalysis.ThrowingTraceListener+DebugAssertFailureException: When created kind       : StaUnitTest
      When created thread id  : 23
      When created thread name: StaTaskScheduler thread
      Current thread id       : 4
      Current thread name     : 
      
         at Microsoft.CodeAnalysis.ThrowingTraceListener.Fail(String message, String detailMessage)
         at System.Diagnostics.TraceListener.Fail(String message)
         at System.Diagnostics.TraceInternal.Fail(String message)
         at System.Diagnostics.Debug.Assert(Boolean condition, String message)
         at Microsoft.CodeAnalysis.Editor.Shared.Utilities.ForegroundThreadAffinitizedObject.AssertIsForeground()
         at Microsoft.CodeAnalysis.Editor.Tagging.AbstractAsynchronousTaggerProvider`1.TagSource.set_CachedTagTrees(ImmutableDictionary`2 value)
         at Microsoft.CodeAnalysis.Editor.Tagging.AbstractAsynchronousTaggerProvider`1.TagSource..ctor(ITextView textViewOpt, ITextBuffer subjectBuffer, AbstractAsynchronousTaggerProvider`1 dataSource, IAsynchronousOperationListener asyncListener, IForegroundNotificationService notificationService)
         at Microsoft.CodeAnalysis.Editor.Tagging.AbstractAsynchronousTaggerProvider`1.GetOrCreateTagSource(ITextView textViewOpt, ITextBuffer subjectBuffer)
         at Microsoft.CodeAnalysis.Editor.Tagging.AbstractAsynchronousTaggerProvider`1.CreateTaggerWorker[T](ITextView textViewOpt, ITextBuffer subjectBuffer)
         at Microsoft.CodeAnalysis.Editor.Tagging.AsynchronousViewTaggerProvider`1.CreateTagger[T](ITextView textView, ITextBuffer subjectBuffer)
         at Microsoft.CodeAnalysis.Editor.Tagging.AsynchronousViewTaggerProvider`1.Microsoft.VisualStudio.Text.Tagging.IViewTaggerProvider.CreateTagger[T](ITextView textView, ITextBuffer buffer)
         at Microsoft.VisualStudio.Text.Tagging.Implementation.TagAggregator`1.GatherTaggers(ITextBuffer textBuffer)
      
      Exception 2:
      Microsoft.CodeAnalysis.ThrowingTraceListener+DebugAssertFailureException: When created kind       : StaUnitTest
      When created thread id  : 23
      When created thread name: StaTaskScheduler thread
      Current thread id       : 4
      Current thread name     : 
      
         at Microsoft.CodeAnalysis.ThrowingTraceListener.Fail(String message, String detailMessage)
         at System.Diagnostics.TraceListener.Fail(String message)
         at System.Diagnostics.TraceInternal.Fail(String message)
         at System.Diagnostics.Debug.Assert(Boolean condition, String message)
         at Microsoft.CodeAnalysis.Editor.Shared.Utilities.ForegroundThreadAffinitizedObject.AssertIsForeground()
         at Microsoft.CodeAnalysis.Editor.Tagging.AbstractAsynchronousTaggerProvider`1.TagSource.set_CachedTagTrees(ImmutableDictionary`2 value)
         at Microsoft.CodeAnalysis.Editor.Tagging.AbstractAsynchronousTaggerProvider`1.TagSource..ctor(ITextView textViewOpt, ITextBuffer subjectBuffer, AbstractAsynchronousTaggerProvider`1 dataSource, IAsynchronousOperationListener asyncListener, IForegroundNotificationService notificationService)
         at Microsoft.CodeAnalysis.Editor.Tagging.AbstractAsynchronousTaggerProvider`1.GetOrCreateTagSource(ITextView textViewOpt, ITextBuffer subjectBuffer)
         at Microsoft.CodeAnalysis.Editor.Tagging.AbstractAsynchronousTaggerProvider`1.CreateTaggerWorker[T](ITextView textViewOpt, ITextBuffer subjectBuffer)
         at Microsoft.CodeAnalysis.Editor.Tagging.AsynchronousViewTaggerProvider`1.CreateTagger[T](ITextView textView, ITextBuffer subjectBuffer)
         at Microsoft.CodeAnalysis.Editor.Tagging.AsynchronousViewTaggerProvider`1.Microsoft.VisualStudio.Text.Tagging.IViewTaggerProvider.CreateTagger[T](ITextView textView, ITextBuffer buffer)
         at Microsoft.VisualStudio.Text.Tagging.Implementation.TagAggregator`1.GatherTaggers(ITextBuffer textBuffer)
      
      Exception 3:
      Microsoft.CodeAnalysis.ThrowingTraceListener+DebugAssertFailureException: When created kind       : StaUnitTest
      When created thread id  : 23
      When created thread name: StaTaskScheduler thread
      Current thread id       : 4
      Current thread name     : 
      
         at Microsoft.CodeAnalysis.ThrowingTraceListener.Fail(String message, String detailMessage)
         at System.Diagnostics.TraceListener.Fail(String message)
         at System.Diagnostics.TraceInternal.Fail(String message)
         at System.Diagnostics.Debug.Assert(Boolean condition, String message)
         at Microsoft.CodeAnalysis.Editor.Shared.Utilities.ForegroundThreadAffinitizedObject.AssertIsForeground()
         at Microsoft.CodeAnalysis.Editor.Tagging.AbstractAsynchronousTaggerProvider`1.TagSource.set_CachedTagTrees(ImmutableDictionary`2 value)
         at Microsoft.CodeAnalysis.Editor.Tagging.AbstractAsynchronousTaggerProvider`1.TagSource..ctor(ITextView textViewOpt, ITextBuffer subjectBuffer, AbstractAsynchronousTaggerProvider`1 dataSource, IAsynchronousOperationListener asyncListener, IForegroundNotificationService notificationService)
         at Microsoft.CodeAnalysis.Editor.Tagging.AbstractAsynchronousTaggerProvider`1.GetOrCreateTagSource(ITextView textViewOpt, ITextBuffer subjectBuffer)
         at Microsoft.CodeAnalysis.Editor.Tagging.AbstractAsynchronousTaggerProvider`1.CreateTaggerWorker[T](ITextView textViewOpt, ITextBuffer subjectBuffer)
         at Microsoft.CodeAnalysis.Editor.Tagging.AsynchronousViewTaggerProvider`1.CreateTagger[T](ITextView textView, ITextBuffer subjectBuffer)
         at Microsoft.CodeAnalysis.Editor.Tagging.AsynchronousViewTaggerProvider`1.Microsoft.VisualStudio.Text.Tagging.IViewTaggerProvider.CreateTagger[T](ITextView textView, ITextBuffer buffer)
         at Microsoft.VisualStudio.Text.Tagging.Implementation.TagAggregator`1.GatherTaggers(ITextBuffer textBuffer)
      
      Exception 4:
      Microsoft.CodeAnalysis.ThrowingTraceListener+DebugAssertFailureException: When created kind       : StaUnitTest
      When created thread id  : 23
      When created thread name: StaTaskScheduler thread
      Current thread id       : 4
      Current thread name     : 
      
         at Microsoft.CodeAnalysis.ThrowingTraceListener.Fail(String message, String detailMessage)
         at System.Diagnostics.TraceListener.Fail(String message)
         at System.Diagnostics.TraceInternal.Fail(String message)
         at System.Diagnostics.Debug.Assert(Boolean condition, String message)
         at Microsoft.CodeAnalysis.Editor.Shared.Utilities.ForegroundThreadAffinitizedObject.AssertIsForeground()
         at Microsoft.CodeAnalysis.Editor.Implementation.EditAndContinue.ActiveStatementTagger.<GetTags>d__11.MoveNext()
         at Microsoft.VisualStudio.Text.Tagging.Implementation.TagAggregator`1.<GetTagsForBuffer>d__39.MoveNext()
      
      
      ---- Microsoft.CodeAnalysis.ThrowingTraceListener+DebugAssertFailureException : When created kind       : StaUnitTest
      When created thread id  : 23
      When created thread name: StaTaskScheduler thread
      Current thread id       : 4
      Current thread name     : 
      
      ---- Microsoft.CodeAnalysis.ThrowingTraceListener+DebugAssertFailureException : When created kind       : StaUnitTest
      When created thread id  : 23
      When created thread name: StaTaskScheduler thread
      Current thread id       : 4
      Current thread name     : 
      
      ---- Microsoft.CodeAnalysis.ThrowingTraceListener+DebugAssertFailureException : When created kind       : StaUnitTest
      When created thread id  : 23
      When created thread name: StaTaskScheduler thread
      Current thread id       : 4
      Current thread name     : 
      
      ---- Microsoft.CodeAnalysis.ThrowingTraceListener+DebugAssertFailureException : When created kind       : StaUnitTest
      When created thread id  : 23
      When created thread name: StaTaskScheduler thread
      Current thread id       : 4
      Current thread name     : 
      
      ---- Microsoft.CodeAnalysis.ThrowingTraceListener+DebugAssertFailureException : When created kind       : StaUnitTest
      When created thread id  : 23
      When created thread name: StaTaskScheduler thread
      Current thread id       : 4
      Current thread name     : 
      
System.InvalidOperationException: Should have been disposed!
DataSource-StackTrace:
   at Microsoft.CodeAnalysis.Editor.Tagging.AbstractAsynchronousTaggerProvider`1..ctor(IAsynchronousOperationListener asyncListener, IForegroundNotificationService notificationService)
   at Microsoft.CodeAnalysis.Editor.Tagging.AsynchronousViewTaggerProvider`1..ctor(IAsynchronousOperationListener asyncListener, IForegroundNotificationService notificationService)
   at Microsoft.CodeAnalysis.Editor.Implementation.BraceMatching.BraceHighlightingViewTaggerProvider..ctor(IBraceMatchingService braceMatcherService, IForegroundNotificationService notificationService, IAsynchronousOperationListenerProvider listenerProvider)
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Microsoft.VisualStudio.Composition.RuntimeExportProviderFactory.RuntimeExportProvider.RuntimePartLifecycleTracker.CreateValue()
   at Microsoft.VisualStudio.Composition.ExportProvider.PartLifecycleTracker.Create()
   at Microsoft.VisualStudio.Composition.ExportProvider.PartLifecycleTracker.MoveNext(PartLifecycleState nextState)
   at Microsoft.VisualStudio.Composition.ExportProvider.PartLifecycleTracker.MoveToState(PartLifecycleState requiredState)
   at Microsoft.VisualStudio.Composition.ExportProvider.PartLifecycleTracker.GetValueReadyToExpose()
   at Microsoft.VisualStudio.Composition.RuntimeExportProviderFactory.RuntimeExportProvider.<>c__DisplayClass15_0.<GetExportedValueHelper>b__0()
   at Microsoft.VisualStudio.Composition.DelegateServices.<>c__DisplayClass2_0`1.<As>b__0()
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.LazyInitValue()
   at System.Lazy`1.get_Value()
   at Microsoft.VisualStudio.Text.Tagging.Implementation.TagAggregator`1.GatherTaggers(ITextBuffer textBuffer)
   at Microsoft.VisualStudio.Text.Tagging.Implementation.TagAggregator`1.<Initialize>b__13_0(ITextBuffer buffer)
   at Microsoft.VisualStudio.Text.Projection.Implementation.BufferGraph.GetTextBuffers(Predicate`1 match)
   at Microsoft.VisualStudio.Text.Tagging.Implementation.TagAggregator`1.Initialize()
   at Microsoft.VisualStudio.Text.Tagging.Implementation.TagAggregator`1.EnsureInitialized()
   at Microsoft.VisualStudio.Utilities.JoinableTaskHelper.<>c__DisplayClass4_0.<<RunOnUIThread>b__0>d.MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.MoveNextRunner.InvokeMoveNext(Object stateMachine)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.MoveNextRunner.Run()
   at Microsoft.VisualStudio.Threading.JoinableTaskFactory.SingleExecuteProtector.TryExecute()
   at Microsoft.VisualStudio.Threading.JoinableTaskFactory.SingleExecuteProtector.<>c.<.cctor>b__20_0(Object state)
   at Xunit.Sdk.AsyncTestSyncContext.<>c__DisplayClass7_0.<Post>b__0()
   at Xunit.Sdk.XunitWorkerThread.<>c.<QueueUserWorkItem>b__4_0(Object _)
   at System.Threading.QueueUserWorkItemCallback.WaitCallback_Context(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()


StackTrace:
   at Microsoft.CodeAnalysis.Editor.Tagging.AbstractAsynchronousTaggerProvider`1.TagSource.DebugRecordInitialStackTrace()
   at Microsoft.CodeAnalysis.Editor.Tagging.AbstractAsynchronousTaggerProvider`1.TagSource..ctor(ITextView textViewOpt, ITextBuffer subjectBuffer, AbstractAsynchronousTaggerProvider`1 dataSource, IAsynchronousOperationListener asyncListener, IForegroundNotificationService notificationService)
   at Microsoft.CodeAnalysis.Editor.Tagging.AbstractAsynchronousTaggerProvider`1.GetOrCreateTagSource(ITextView textViewOpt, ITextBuffer subjectBuffer)
   at Microsoft.CodeAnalysis.Editor.Tagging.AbstractAsynchronousTaggerProvider`1.CreateTaggerWorker[T](ITextView textViewOpt, ITextBuffer subjectBuffer)
   at Microsoft.CodeAnalysis.Editor.Tagging.AsynchronousViewTaggerProvider`1.CreateTagger[T](ITextView textView, ITextBuffer subjectBuffer)
   at Microsoft.CodeAnalysis.Editor.Tagging.AsynchronousViewTaggerProvider`1.Microsoft.VisualStudio.Text.Tagging.IViewTaggerProvider.CreateTagger[T](ITextView textView, ITextBuffer buffer)
   at Microsoft.VisualStudio.Text.Tagging.Implementation.TagAggregator`1.GatherTaggers(ITextBuffer textBuffer)
   at Microsoft.VisualStudio.Text.Tagging.Implementation.TagAggregator`1.<Initialize>b__13_0(ITextBuffer buffer)
   at Microsoft.VisualStudio.Text.Projection.Implementation.BufferGraph.GetTextBuffers(Predicate`1 match)
   at Microsoft.VisualStudio.Text.Tagging.Implementation.TagAggregator`1.Initialize()
   at Microsoft.VisualStudio.Text.Tagging.Implementation.TagAggregator`1.EnsureInitialized()
   at Microsoft.VisualStudio.Utilities.JoinableTaskHelper.<>c__DisplayClass4_0.<<RunOnUIThread>b__0>d.MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.MoveNextRunner.InvokeMoveNext(Object stateMachine)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.MoveNextRunner.Run()
   at Microsoft.VisualStudio.Threading.JoinableTaskFactory.SingleExecuteProtector.TryExecute()
   at Microsoft.VisualStudio.Threading.JoinableTaskFactory.SingleExecuteProtector.<>c.<.cctor>b__20_0(Object state)
   at Xunit.Sdk.AsyncTestSyncContext.<>c__DisplayClass7_0.<Post>b__0()
   at Xunit.Sdk.XunitWorkerThread.<>c.<QueueUserWorkItem>b__4_0(Object _)
   at System.Threading.QueueUserWorkItemCallback.WaitCallback_Context(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()

   at Roslyn.Utilities.Contract.Fail(String message)
   at Microsoft.CodeAnalysis.Editor.Tagging.AbstractAsynchronousTaggerProvider`1.TagSource.Finalize()
```

**Comment by dpoeschl** [comment]
There's at least 3 features in there dying (EnC, Rename Tracking, Tagging) because of `ForegroundThreadAffinitizedObject.AssertIsForeground()` failing. There have been lots of these. https://github.com/dotnet/roslyn/issues?q=is%3Aissue+AssertIsForeground+is%3Aclosed in the past and at one point thought it might have been fixed by  https://github.com/dotnet/roslyn/pull/18837 but apparently not.

**Comment by dpoeschl** [comment]
Since this is a general foreground-thread-in-tests problem, removing my assignment and forwarding to @jinujoseph for triage.

**Comment by JieCarolHu** [comment]
Failed again today in dev15.7.x windows_debug_unit32 #196



When created kind       : StaUnitTest
When created thread id  : 9
When created thread name: StaTaskScheduler thread
Current thread id       : 18
Current thread name     : 

   at Microsoft.CodeAnalysis.ThrowingTraceListener.Fail(String message, String detailMessage) in /_/src/Test/Utilities/Portable/ThrowingTraceListener.cs:line 24
   at System.Diagnostics.TraceListener.Fail(String message) in f:\dd\NDP\fx\src\compmod\system\diagnostics\TraceListener.cs:line 180
   at System.Diagnostics.TraceInternal.Fail(String message) in f:\dd\NDP\fx\src\compmod\system\diagnostics\TraceInternal.cs:line 225
   at System.Diagnostics.Debug.Assert(Boolean condition, String message) in f:\dd\NDP\fx\src\compmod\system\diagnostics\Debug.cs:line 108
   at Microsoft.CodeAnalysis.Editor.Shared.Utilities.ForegroundThreadAffinitizedObject.AssertIsForeground() in /_/src/EditorFeatures/Core/Shared/Utilities/ForegroundThreadAffinitizedObject.cs:line 105
   at Microsoft.CodeAnalysis.Editor.Implementation.RenameTracking.RenameTrackingTaggerProvider.StateMachine.Connect() in /_/src/EditorFeatures/Core/Implementation/RenameTracking/RenameTrackingTaggerProvider.StateMachine.cs:line 366
   at Microsoft.CodeAnalysis.Editor.Implementation.RenameTracking.RenameTrackingTaggerProvider.Tagger..ctor(StateMachine stateMachine, ITextUndoHistoryRegistry undoHistoryRegistry, IWaitIndicator waitIndicator, IEnumerable`1 refactorNotifyServices) in /_/src/EditorFeatures/Core/Implementation/RenameTracking/RenameTrackingTaggerProvider.Tagger.cs:line 33
   at Microsoft.CodeAnalysis.Editor.Implementation.RenameTracking.RenameTrackingTaggerProvider.CreateTagger[T](ITextBuffer buffer) in /_/src/EditorFeatures/Core/Implementation/RenameTracking/RenameTrackingTaggerProvider.cs:line 64
   at Microsoft.VisualStudio.Text.Tagging.Implementation.TagAggregator`1.GatherTaggers(ITextBuffer textBuffer) in E:\A\_work\39\s\src\Editor\Text\Impl\TagAggregator\TagAggregator.cs:line 609

**Comment by sharwell** [comment]
@JieCarolHu That last issue is fixed by #25681
